src/databricks/sql/utils.py

```diff
@@ -542,20 +542,41 @@ def infer_types(params: list[DbSqlParameter]):
         datetime.date: DbSqlType.DATE,
         bool: DbSqlType.BOOLEAN,
         Decimal: DbSqlType.DECIMAL,
+        type(None): DbSqlType.VOID,
```

Comment:
This is mostly something that should have been mentioned earlier, but this lookup stable is static, so pull it to an outer layer (possibly the outermost layer) so that it's not created on every call.